octodns_azure/__init__.py

```diff
@@ -584,14 +584,18 @@ class AzureBaseProvider(BaseProvider):
         ('A', 'AAAA', 'CAA', 'CNAME', 'MX', 'NS', 'PTR', 'SRV', 'TXT')
     )
 
+    CREDENTIAL_METHOD_CLIENT_SECRET = "client_secret"
+    CREDENTIAL_METHOD_CLI = "cli"
+
     def __init__(
         self,
         id,
-        client_id,
-        key,
-        directory_id,
         sub_id,
         resource_group,
+        directory_id="",
+        client_id="",
+        key="",
```

Comment:
Would be more consistent with other default params elsewhere to use `None` for this over `""`

```suggestion
        directory_id=None,
        client_id=None,
        key=None,
```

Reply:
My linter was complaining that these are expected to be strings (but python is loosely typed). I'lll update that.